corgi/tasks/monitoring.py

```
@@ -94,7 +95,7 @@ def email_failed_tasks():
 
     failed_tasks_threshold = get_last_success_for_task("corgi.tasks.monitoring.email_failed_tasks")
     failed_tasks = TaskResult.objects.filter(
-        status__exact="FAILURE",
+        status__in=CELERY_TASK_EXCEPTION_STATES,
```

I'm also not sure that we should report retried tasks, but I'm probably not understanding the Celery task states. I thought that we'd create a RETRY TaskResult for any retried attempts, and a SUCCESS / FAILURE TaskResult for the final attempt once the maximum number of retries has been reached. In that case, wouldn't we get a lot of email spam for tasks which had some temporary error but retried successfully?

Or if that's wrong and there's only one TaskResult, does its status change to SUCCESS / FAILURE at the end of the retry loop? In that case, our existing monitoring emails should report any failures, right? I'm confused how a TaskResult ended up in the RETRY state. The number of retries should be limited - it's not an infinite retry loop I hope (or if it is, let's fix that bug).

---

Discussed in Gchat / standup - we're not sure where the RETRY tasks come from, but something weird is happening with the TaskResults so reporting these until we fix that other bug LGTM.